src/Identifier/Identifier.php

```diff
@@ -15,13 +21,31 @@ class Identifier
      */
     private $type;
 
+    /**
+     * @param string $name
+     * @param IdentifierType $type
+     *
+     * @throws InvalidIdentifierName
+     */
     public function __construct(string $name, IdentifierType $type)
     {
         $this->type = $type;
 
+        if (self::WILDCARD === $name
+            || ReflectionFunctionAbstract::CLOSURE_NAME === $name
+            || 0 === \strpos($name, ReflectionClass::ANONYMOUS_CLASS_NAME_PREFIX)
+        ) {
+            $this->name = $name;
+            return;
+        }
+
         $name = \ltrim($name, '\\');
-        // @todo validate the name somehow (see issue #20)
-        $this->name = (string) $name;
+
+        if ( ! \preg_match('/([a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)(\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)*/', $name)) {
```

Comment:
Private constant plz :-)

Reply:
This pattern makes `\\` valid - At least the first block should be a `+` instead of a `*`

Reply:
No, there's still `[a-zA-Z_\x7f-\xff]` at the beginning.

Reply:
Dang, I need some sleep @_@